src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -43,16 +47,40 @@ public partial class Startup
     {
         static Startup()
         {
-            Triggers<BuildChannel>.Inserted += entry =>
+            Triggers<BuildChannel>.Inserting += entry =>
             {
+                BuildAssetRegistryContext context = entry.Context as BuildAssetRegistryContext;
                 BuildChannel entity = entry.Entity;
+                if (ChannelHasAssociatedReleasePipeline(entity.ChannelId, context))
+                {
+                    entry.Cancel = true;
+                    var queue = context.GetService<BackgroundQueue>();
+                    var releasePipelineRunner = context.GetService<IReleasePipelineRunner>();
+                    queue.Post(() => releasePipelineRunner.StartAssociatedReleasePipelinesAsync(entity.BuildId, entity.ChannelId));
+                }
+            };
+
+            Triggers<BuildChannel>.Inserted += entry =>
+            {
                 DbContext context = entry.Context;
+                BuildChannel entity = entry.Entity;
+
                 var queue = context.GetService<BackgroundQueue>();
                 var dependencyUpdater = context.GetService<IDependencyUpdater>();
                 queue.Post(() => dependencyUpdater.StartUpdateDependenciesAsync(entity.BuildId, entity.ChannelId));
             };
         }
 
+        private static bool ChannelHasAssociatedReleasePipeline(int channelId, BuildAssetRegistryContext context)
+        {
+            Channel channel = context.Channels
+                .Where(ch => ch.Id == channelId)
+                .Include(ch => ch.ChannelReleasePipelines)
+                .ThenInclude(crp => crp.ReleasePipeline)
+                .FirstOrDefault();
```

Comment:
It might be clearer to just put the condition in the FirstOrDefault:

return context.Channels
                .Where(ch => ch.Id == channelId)
                .Include(ch => ch.ChannelReleasePipelines)
                .ThenInclude(crp => crp.ReleasePipeline)
                .FirstOrDefault(c => c.ChannelReleasePipelines.Count > 0);

Reply:
I like that.